plugins/azure/resoto_plugin_azure/resource/compute.py

```diff
@@ -2582,6 +2632,27 @@ class AzureVirtualMachine(AzureResource, BaseInstance):
     virtual_machine_scale_set: Optional[str] = field(default=None, metadata={"description": ""})
     vm_id: Optional[str] = field(default=None, metadata={'description': 'Specifies the vm unique id which is a 128-bits identifier that is encoded and stored in all azure iaas vms smbios and can be read using platform bios commands.'})  # fmt: skip
 
+    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
+        if placement_group_id := self.proximity_placement_group:
+            builder.add_edge(
+                self, edge_type=EdgeType.default, clazz=AzureProximityPlacementGroup, id=placement_group_id
+            )
+
+        if (
+            self.virtual_machine_storage_profile
+            and self.virtual_machine_storage_profile.image_reference
+            and (image_reference_id := self.virtual_machine_storage_profile.image_reference.id)
+        ):
+            builder.add_edge(self, edge_type=EdgeType.default, clazz=AzureImage, id=image_reference_id)
+
+        if (
+            self.virtual_machine_storage_profile
+            and self.virtual_machine_storage_profile.os_disk
+            and self.virtual_machine_storage_profile.os_disk.managed_disk
+            and (managed_disk_id := self.virtual_machine_storage_profile.os_disk.managed_disk.id)
+        ):
```

Comment:
```suggestion
        if (
            (sp := self.virtual_machine_storage_profile)
            and (disk := sp.os_disk)
            and (managed := disk.managed_disk)
            and (managed_disk_id := managed.id)
        ):
```
If you want to get rid of long chains, you could assign the result to an intermediate variable.